File: ts/src/test/Exchange/test.fetchLiquidations.ts
import assert from 'assert';
import { Exchange } from "../../../ccxt";
import testSharedMethods from './base/test.sharedMethods.js';
import testLiquidation from './base/test.liquidation.js';

async function testFetchLiquidations (exchange: Exchange, skippedProperties: object, code: string) {
    const method = 'fetchLiquidations';
    if (!exchange.has['fetchLiquidations']) {
        return true;
    }
    const items = await exchange.fetchLiquidations (code);
    assert (Array.isArray (items), exchange.id + ' ' + method + ' ' + code + ' must return an array. ' + exchange.json (items));
    // const now = exchange.milliseconds ();
    for (let i = 0; i < items.length; i++) {
        testLiquidation (exchange, skippedProperties, method, items[i], code);
    }
    testSharedMethods.assertTimestampOrder (exchange, method, code, items);
    return true;
}

export default testFetchLiquidations;

File: julia/Ccxt/test/exchange/test.fetchLiquidations.jl
using Test
using Ccxt
function testFetchLiquidations(exchange, skippedProperties, code)

    method = "fetchLiquidations";
    if functions.ccxtruthy(!functions.ccxtruthy(get(get(exchange, Symbol("has"), nothing), Symbol("fetchLiquidations"), nothing)))
            return true
    end
    items = Base.fetch(fetchLiquidations(exchange, code));
    @test functions.ccxtruthy(functions.ccxt_isArray(items))
    i = 0
    while functions.ccxtruthy(functions.ccxt_lt(i, length(items)))
        testLiquidation(exchange, skippedProperties, method, get(items, i + 1, nothing), code);
        i += 1
    end
    assertTimestampOrder(testSharedMethods, exchange, method, code, items);
    return true
end
